core/arch/arm/kernel/boot.c

```diff
@@ -1471,8 +1515,30 @@ void __weak boot_save_args(unsigned long a0, unsigned long a1 __unused,
 	 * here. This is also called Manifest DT, related to the Manifest DT
 	 * passed in the FF-A Boot Information Blob, but with a different
 	 * compatible string.
+
+	 * Scenario B: FW Handoff via Transfer List
+	 * Note: FF-A is not yet supported with Transfer List
+	 * a0	- DTB address or 0 (AArch64)
+	 *	- must be 0 (AArch32)
+	 * a1	- TRANSFER_LIST_SIGNATURE | REG_CONVENTION_VER_MASK
+	 * a2	- must be 0 (AArch64)
+	 *	- DTB address or 0 (AArch32)
+	 * a3	- Transfer list base address
+	 * a4	- Not used
```

Comment:
When `CFG_WITH_ARM_TRUSTED_FW=n` (Armv7), a4 hold Non-secure entry address. Should this be described here or is does firmware handoff transfer list supported only on Armv8+?

Reply:
The TL is supposed to support both v7 and v8. I believe we need to define a new TE for non-secure entry later.
Currently we can only support the defined TEs described in in the spec.
I will add one more comment here to note that currently non-secure entry is not yet supported.